scripts/viewer.js

```diff
@@ -0,0 +1,150 @@
+      }
+      bracketAddresses = argv.brackets
+    }
+    const exchange = await exchangePromise
+
+    const detailedBrackets = await Promise.all(
+      bracketAddresses.map(async (bracketAddress) => {
+        const orders = decodeOrders(await exchange.getEncodedUserOrders.call(bracketAddress))
+        let tradedTokenIds = []
+        for (const order of orders) {
+          tradedTokenIds.push(order.buyToken, order.sellToken)
+        }
+        tradedTokenIds = uniqueItems(tradedTokenIds)
+        const tokenInfoPromises = fetchTokenInfoFromExchange(exchange, tradedTokenIds)
+        return {
+          bracketAddress,
+          tokenIds: tradedTokenIds,
+          tokenInfoPromises,
+        }
+      })
+    )
+    console.log("Recovering token balances per bracket...")
+    const tokenBalancesPerUser = await Promise.all(
+      detailedBrackets.map(async ({ bracketAddress, tokenIds, tokenInfoPromises }) => {
+        const tokenBalances = []
+        for (const tokenId of tokenIds) {
+          const tokenData = await tokenInfoPromises[tokenId]
+          const [availableForTrading, requestedForWithdraw, storedInBracket] = await Promise.all([
+            exchange.getBalance(bracketAddress, tokenData.address),
+            getWithdrawableAmount(bracketAddress, tokenData.address, exchange, web3),
+            tokenData.instance.balanceOf(bracketAddress),
+          ])
+          tokenBalances.push({
+            bracketAddress: bracketAddress,
+            tokenId,
+            tokenData,
+            totalBalance: availableForTrading.add(requestedForWithdraw).add(storedInBracket),
+            availableForTrading,
+            requestedForWithdraw,
+            storedInBracket,
+          })
+        }
+        return tokenBalances
+      })
+    )
+
+    const balances = []
+    for (const tokenBalances of tokenBalancesPerUser) {
+      balances.push(...tokenBalances)
+    }
+
+    const totalBalanceSum = {}
+    const requestedForWithdrawSum = {}
+    const storedInBracketSum = {}
+    for (const balanceItem of balances) {
+      if (balanceItem.totalBalance.isZero()) {
+        continue
+      }
+      totalBalanceSum[balanceItem.tokenId] = (totalBalanceSum[balanceItem.tokenId] || new BN(0)).add(balanceItem.totalBalance)
+      requestedForWithdrawSum[balanceItem.tokenId] = (requestedForWithdrawSum[balanceItem.tokenId] || new BN(0)).add(
+        balanceItem.requestedForWithdraw
+      )
+      storedInBracketSum[balanceItem.tokenId] = (storedInBracketSum[balanceItem.tokenId] || new BN(0)).add(
+        balanceItem.storedInBracket
+      )
+      const tokenData = balanceItem.tokenData
+      const optionalString = buildOptionalString({ tokenData, ...balanceItem })
+      console.log(
+        `Balance of ${balanceItem.bracketAddress} for token ${tokenData.symbol}: ${fromErc20Units(
+          balanceItem.totalBalance,
+          tokenData.decimals
+        )}${optionalString}`
+      )
+    }
+
+    console.log("/nTotal funds:")
```

Comment:
It doesn't render as intended in my Mac OS terminal, see screen:
<img width="107" alt="Screenshot 2020-07-01 at 12 17 27" src="https://user-images.githubusercontent.com/414573/86226805-d9b08680-bb94-11ea-9866-492a4a6af203.png">

Reply:
Should be fixed now, let me know if this is still an issue for you.